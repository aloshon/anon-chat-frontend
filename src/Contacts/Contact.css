.contact-card.card {
    position: relative;
    top: 25px;
    margin: 15px;
}

@media (min-width: 1000px) {
    /* .group-chat-card {
        position: relative;
        display: inline-block;
        width: 49%;
        vertical-align: top;
    } */

    .contact-card.card {
        display:inline-block;
        width: 47%;
        padding-bottom: 20px;
        vertical-align: top;
        margin-left: 10px;
    }
}