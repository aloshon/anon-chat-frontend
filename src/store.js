import  { composeWithDevTools } from "redux-devtools-extension";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";

import thunk from "redux-thunk";
import rootReducer from "./Reducers/rootReducer";
import { createStore, applyMiddleware } from "redux";

// only keep the darkMode reducer persistent in local storage
const persistConfig = {
  key: "root",
  whitelist:['darkMode'],
  storage,
  stateReconciler: autoMergeLevel2
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(
    applyMiddleware(thunk),
  )
);

export const persistedStore = persistStore(store);