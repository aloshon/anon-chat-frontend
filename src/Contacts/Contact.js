import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import {useSelector, useDispatch} from 'react-redux';
import {removeContact} from '../Actions/actionCreators';
import "./Contact.css";

const Contact = ({userId, nickname, username}) => {
    const darkMode = useSelector(state => state.darkMode);
    const dispatch = useDispatch();

    return (
        <Card className="contact-card" style={{backgroundColor: darkMode.card}}>
            <Card.Body>
                <Card.Title>{nickname}</Card.Title>
                <small><Card.Text>Username: {username}</Card.Text></small>
                <Button variant="danger" onClick={() => removeContact(dispatch, userId)}>X</Button>
            </Card.Body>
        </Card>
    )
}

export default Contact;