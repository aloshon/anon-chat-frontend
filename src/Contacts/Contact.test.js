import { render, screen } from '@testing-library/react';
import Contact from './Contact';
import { Provider } from "react-redux";
import { store } from "../store"
import {BrowserRouter} from 'react-router-dom';


describe("Contact should render username", () => {
    const {asFragment} = render(
      <Provider store={store}>
        <BrowserRouter>
            <Contact 
                userId={1}
                nickname={"test"}
                username={"user1"}
            />
        </BrowserRouter>
      </Provider>
    );
    test("nickname and username are displayed", () => {
      const username = screen.getByText("Username: user1");
      expect(username).toBeInTheDocument();
      const nickname = screen.getByText("test");
      expect(nickname).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
  });