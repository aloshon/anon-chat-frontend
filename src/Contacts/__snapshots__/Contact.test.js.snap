// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should render username matches snapshot 1`] = `<DocumentFragment />`;
